lens/util.go

```diff
@@ -0,0 +1,58 @@
+package lens
+
+import (
+	"context"
+	"errors"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/chain/state"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/node/impl/full"
+	cbor "github.com/ipfs/go-ipld-cbor"
+	logging "github.com/ipfs/go-log/v2"
+	"golang.org/x/xerrors"
+)
+
+var logger = logging.Logger("visor/lens/lotus")
+
+// OptimizedStateGetActorWithFallback is a helper to obtain an actor in the
+// state of the current tipset without recomputing the full tipset. It does
+// this by obtaining the child tipset (current height+1) and using the
+// pre-computed ParentState().
+//
+// TODO: Remove. See:  https://github.com/filecoin-project/sentinel-visor/issues/196
+func OptimizedStateGetActorWithFallback(ctx context.Context, store cbor.IpldStore, chainAPI full.ChainModuleAPI, fallback full.StateModuleAPI, actor address.Address, tsk types.TipSetKey) (*types.Actor, error) {
```

Comment:
This could be simpler if `FullNode` actually implemented `FullNodeAPI` but it doesn't:

```
lens/util/repo.go:109:74: cannot use ra.FullNodeAPI (type impl.FullNodeAPI) as type api.FullNode in argument to lens.OptimizedStateGetActorWithFallback:
	impl.FullNodeAPI does not implement api.FullNode (AuthNew method has pointer receiver)
```